multi arc gauge enhancement to draw more than one arc in gauges

GoekhanOezen opened the pull request:
See issue #1077.

For each `data_column` one arc, the related background arc, arc label and a line to highlight the relation from arc to label will be drawn. When there are multiple `data_column` the legend is shown. For cleaner calculation in functions some variables are added. When there only is one `data_column` visible the old behaviour didnt changed.

This is my first pull request. We are already using the feature. I spend much time to refactor but i am sure there are still some optimiziations left.

All existing tests passed.
I would be glad to hear your opinion to the changes.

![multi arc gauge 1](https://cloud.githubusercontent.com/assets/11871137/9543671/d440d782-4d7a-11e5-85de-bbcfe2605c02.png)
![multi arc gauge 2](https://cloud.githubusercontent.com/assets/11871137/9543674/d579c06e-4d7a-11e5-8db3-3678c9d81058.png)
![multi arc gauge 3](https://cloud.githubusercontent.com/assets/11871137/9543673/d4b3765c-4d7a-11e5-9ce9-b935f8927826.png)

chocolatemurderer commented:
hi there, will you be merging this soon? I would really love this feature as well but currently compiling it myself...

Tropicalista commented:
+1 Seems a really good feature.

aendrew commented:
This looks amazing! Would love to add it, but it seriously needs some unit tests — we're not adding any new features to 0.4.x without them, especially when it has this many LOC.

Could you please do that, @GoekhanOezen? Thanks!

GoekhanOezen commented:
I will try but I would appreciate every help from the community.

aendrew commented:
@GoekhanOezen Try using the tests in [spec/arc-spec.js](https://github.com/c3js/c3/blob/master/spec/arc-spec.js) as a guide. Seriously, the screenshots look amazing and this would make gauge charts much cooler than they presently are! Do let me know if you need any help, am usually on [Gitter](http://gitter.im/c3js/c3). 😄

aendrew commented on src/data.js:
```diff
@@ -176,9 +176,11 @@ c3_chart_internal_fn.mapTargetsToUniqueXs = function (targets) {
 };
 c3_chart_internal_fn.addHiddenTargetIds = function (targetIds) {
     this.hiddenTargetIds = this.hiddenTargetIds.concat(targetIds);
```
This line light conflict with #1662...

GoekhanOezen replied:
shall I wait #1663 to be merged. Because I would place my line in the if statement. I also dont want to increase or decrease `this.hiddenTargetCount` if not every ID in `targetIds` is added to `this.hiddenTargetIds`

aendrew replied:
@GoekhanOezen #1663 is now merged; mind updating this and rebasing? Thanks!

GoekhanOezen commented:
ok i have rebased, refactored, bugfixed, refactored and added some test. i fixed the warnings and the tests who are related to my changes. But i cannot understand the error shown in travis. Because everytime i run tests on my PC there are different error displayed. i have a screenshot added below. I didnt change anything on axis.js. 
Can someone help pls?


![grafik](https://user-images.githubusercontent.com/11871137/27485766-582ae348-582e-11e7-8924-1212752b5baa.png)

micschk commented:
Would be great if this feature could be merged!

pms130030 commented:
Would love this feature in c3! I wish I knew enough about this to help finish it

kt3k commented on spec/arc-spec.js:
```diff
@@ -193,6 +193,99 @@ describe('c3 chart arc', function () {
+
+                });
+
+                it('if only one data_column is visible the label should have "" for transform', function () {
```
@GoekhanOezen
`done` is missing here and that causes uncaught error in a following test case which is running 1 second later of this test case.

kt3k commented:
Landed [here](https://github.com/c3js/c3/commit/3e635a0e93e03381f72132c3c2dcaeee0c6ce2fd) in master branch.

Released in [v0.4.21](https://github.com/c3js/c3/releases/tag/v0.4.21). Thanks for the big contribution. Sorry for the serious delay!

causztic commented:
thanks for the efforts! 🎊

srinivasyl commented:
@GoekhanOezen  can we provide two different max values for each data source like you've data1 and data2. I want to set 100 as max value for data1 and 200 as max value for data2 is that possbile in c3.js library. how to achieve it?
